torch/linalg/__init__.py

```diff
@@ -344,49 +381,78 @@
-for real-valued :attr:`input`, or the conjugate transpose of `V` for complex-valued :attr:`input`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{\text{H}}
+
+where :math:`V^{\text{H}}` denotes the conjugate transpose in the complex case and the transpose in
+the real case. The matris :math:`V` (and thus :math:`V^{\text{H}}`)
+is orthogonal in the real case, and unitary in the complex case.
+
+The decomposition is represented as a namedtuple `(eigenvalues, eigenvectors)`, which
+corresepond to `(L, V)` above. If :attr:`input` is a batch of matrices, then `L`, `V` are also
+batched with the same batch dimensions.
+
+The eigenvalues are returned in ascending order.
 
-Since the matrix or matrices in :attr:`input` are assumed to be Hermitian, the imaginary part of their diagonals
-is always treated as zero. When :attr:`UPLO` is "L", its default value, only the lower triangular part of each
-matrix is used in the computation. When :attr:`UPLO` is "U" only the upper triangular part of each matrix is used.
+:attr:`input` is assumed to be Hermitian (resp. symmetric), but this is not checked internally.
+When :attr:`UPLO` is `'L'` (default), only the lower triangular part of each matrix is used in the computation.
+When :attr:`UPLO` is `'U'`, only the upper triangular part of each matrix is used.
 
-Supports input of float, double, cfloat and cdouble dtypes.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+The eigenvalues will always be real-valued, even if :attr:`input` is complex.
+The eigenvectors will have the same dtype as :attr:`input`.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
 
-.. note:: The eigenvalues/eigenvectors are computed using LAPACK's `syevd` and `heevd` routines for CPU inputs,
-          and MAGMA's `syevd` and `heevd` routines for CUDA inputs.
+.. note:: This function is computed using LAPACK's and MAGMA's `syevd` and `heevd` routines for
+          CPU and CUDA symmetric and Hermitian inputs respectively.
 
 .. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
 
-.. note:: The eigenvectors of matrices are not unique, so any eigenvector multiplied by a constant remains
-          a valid eigenvector. This function may compute different eigenvector representations on
-          different device types. Usually the difference is only in the sign of the eigenvector.
+.. warning:: The eigenvectors of a matrix are not unique. Any eigenvector may be multiplied by `-1`
+             in the real case or by :math:`e^{i \phi}` for any :math:`\phi \in \mathbb{R}` in the
+             complex case.
+             As such, the eigenvectors are not continuous functions of the inputs, and different
+             platforms, like NumPy, or inputs on different devices, may produce different
```

Comment:
I feel like this is the heart of this warning, which should probably be downgraded to a note. Warnings are more appropriate for when something is likely to bite a user. Here this is just a note that different platforms might produce different eigenvectors because eigenvectors are not unique (and there's no canonical set of eigenvectors).

Reply:
I figured that the lack of continuity is something weird, and that most people could think that these maps are continuous, hence the warning.

I'd be happy to downgrade this to a note, if you think this is not too problematic, should we do that?